source/includes/inkling2-reference/_functions.html.md

```diff
@@ -0,0 +1,233 @@
+
+```inkling2--syntax
+variableDeclaration :==
+  var <varName> [':' <varType>] ['=' <expression>]
+```
+
+A variable that is not assigned an initial value must be assigned a value along all code paths before it is used.
+
+```inkling2--code
+# Variable with no initial assignment.
+var TwoOrThree: number<2, 3>
+
+if IsCondition() {
+  TwoOrThree = 2
+}
+
+# This will generate a compiler error because the
+# variable may not have a defined value.
+return TwoOrThree
+```
+
+#### Variable Assignments
+A previously-declared variable can be assigned a new value as long as that value matches the type of the variable.
+
+```inkling2--syntax
+variableAssignment :==
+  <varName> '=' <expression>
+```
+
+#### Conditional Execution
+Inkling supports if/else statements, which can be chained together with “else if” clauses. The braces that surround the code blocks within an `if` or `else` clause are required.
+
+```inkling2--syntax
+conditionalStatement :==
+  if <expression> '{'
+    [functionStatement]*
+  '}' [else if <expression> '{'
+    [functionStatement]*
+  '}']* [else '{'
+    [functionStatement]*
+  '}']
+```   
+
+#### Return Statements
+A return statement terminates execution of the function and returns the specified value to its caller. The type of the expression must match the declared return type of the function.
+
+```inkling2--syntax
+returnStatement :==
+  return <expression>
+```
+
+### Expressions
+Inkling supports basic expressions and a variety of unary and binary operators.
+
+#### Built-in Operators
+| Operation        | Syntax        | Description |
+| -                | -             | -           |
+| Function call    | X()           | Invokes a function with zero or more arguments |
+| Array index      | X[Y]          | Indexes into an array |
+| Member access    | X.Y           | References a field within a structure |
+| Neg/pos          | -, +          | Negate numeric value |
+| Logical not      | not           | Logical not (produces 0 or 1) |
+| Exponent         | **            | X raised to the Y power |
+| Mult/div/mod     | *, /, %       | X times Y, X divided by Y, X mod Y |
+| Add/subtract     | +, -          | X plus Y, X minus Y |
+| Relational       | <, <=, >, >=  | Less, less-or-equal, greater, greater-or-equal |
+| Equality         | ==, !=        | Equal, not equal |
+| Logical and      | and           | Left expression if truthy, else right expression |
+| Logical or       | or            | Left expression if falsy, else right expression |
+
+
+As in algebra, parentheses can be used to make order of evaluation explicit. If parentheses are omitted, the table above defines the order of operations.
+
+All mathematical, logical and relational operators work only with numerical operands. Operands of other types (strings, arrays, structures, etc.) are flagged as errors by the compiler.
+
```

Comment:
The underlying numeric type is float32, right? We should probably mention that in the context of the operators. And we should define the overflow behavior -- will we get Infs and NaNs?

Reply:
No, the underlying numeric type ("number") in Inkling equates to float64. That's explained in the types reference docs already. 

The AI engine uses float32 for NN weights even if the inputs to the NN are higher precision. That's probably what you're thinking of here.